src/category_theory/limits/presheaf.lean

```diff
@@ -0,0 +1,216 @@
+is, an extension of `A : C ⥤ ℰ` to `(Cᵒᵖ ⥤ Type u) ⥤ ℰ` through `yoneda : C ⥤ Cᵒᵖ ⥤ Type u`.
+TODO: Show `colimit_adj.L` is unique amongst cocontinuous functors with this property.
+
+## Tags
+colimit, representable, presheaf
+-/
+
+namespace category_theory
+
+noncomputable theory
+
+open category limits
+universes u₁ u₂
+
+variables {C : Type u₁} [small_category C]
+variables {ℰ : Type u₂} [category.{u₁} ℰ]
+variable (A : C ⥤ ℰ)
+
+namespace colimit_adj
+
+/--
+The functor taking `(E : ℰ) (c : Cᵒᵖ)` to the homset `(A.obj C ⟶ E)`. It is shown in `L_adjunction`
+that this functor has a left adjoint (provided `E` has colimits) given by taking colimits over
+categories of elements.
+In the case where `ℰ = Cᵒᵖ ⥤ Type u` and `A = yoneda`, this functor is isomorphic to the identity.
+
+Defined as in [MM92], Chapter I, Section 5, Theorem 2.
+-/
+@[simps {rhs_md := semireducible}]
+def restricted_yoneda : ℰ ⥤ (Cᵒᵖ ⥤ Type u₁) :=
+yoneda ⋙ (whiskering_left _ _ (Type u₁)).obj (functor.op A)
+
+/--
+The functor `restricted_yoneda` is isomorphic to the identity functor if evaluated at the
+yoneda embedding.
+-/
+def right_is_id : restricted_yoneda (yoneda : C ⥤ Cᵒᵖ ⥤ Type u₁) ≅ 𝟭 _ :=
+nat_iso.of_components
+(λ P, nat_iso.of_components (λ X, yoneda_sections_small X.unop _)
+  (λ X Y f, funext $ λ x,
+  begin
+    dsimp [ulift_trivial, yoneda_lemma],
+    rw ← functor_to_types.naturality _ _ x f (𝟙 _),
+    dsimp,
+    simp,
+  end))
+(λ _ _ _, rfl)
+
+/--
+(Implementation). The equivalence of homsets which helps construct the left adjoint to
+`colimit_adj.restricted_yoneda`.
+It is shown in `Le'_natural` that this is a natural bijection.
+-/
+def Le' (P : Cᵒᵖ ⥤ Type u₁) (E : ℰ) {c : cocone ((category_of_elements.π P).left_op ⋙ A)}
+  (t : is_colimit c) : (c.X ⟶ E) ≃ (P ⟶ (restricted_yoneda A).obj E) :=
+(t.hom_iso' E).to_equiv.trans
+{ to_fun := λ k,
+  { app := λ c p, k.1 (opposite.op ⟨_, p⟩),
+    naturality' := λ c c' f, funext $ λ p,
+      (k.2 (has_hom.hom.op ⟨f, rfl⟩ :
+              (opposite.op ⟨c', P.map f p⟩ : P.elementsᵒᵖ) ⟶ opposite.op ⟨c, p⟩)).symm },
+  inv_fun := λ τ,
+  { val := λ p, τ.app p.unop.1 p.unop.2,
+    property := λ p p' f,
+    begin
+      simp_rw [← f.unop.2],
+      apply (congr_fun (τ.naturality f.unop.1) p'.unop.2).symm,
+    end },
+  left_inv :=
+  begin
+    rintro ⟨k₁, k₂⟩,
+    ext,
+    dsimp,
+    congr' 1,
+    simp,
+  end,
+  right_inv :=
+  begin
+    rintro ⟨_, _⟩,
+    refl,
+  end }
+
+/-- (Implementation). Show that the bijection in `Le'` is natural (on the right). -/
+lemma Le'_natural (P : Cᵒᵖ ⥤ Type u₁) (E₁ E₂ : ℰ) (g : E₁ ⟶ E₂)
+  {c : cocone _} (t : is_colimit c) (k : c.X ⟶ E₁) :
+Le' A P E₂ t (k ≫ g) = Le' A P E₁ t k ≫ (restricted_yoneda A).map g :=
+begin
+  ext _ X p,
+  apply (assoc _ _ _).symm,
+end
+
+variables [has_colimits ℰ]
+
+/-- The left adjoint to the functor `restricted_yoneda`. -/
+def L : (Cᵒᵖ ⥤ Type u₁) ⥤ ℰ :=
```

Comment:
What should we call this?

Reply:
I've added to the docstring that this is an extension of `A` along `yoneda`, so maybe `extend_along_yoneda`? And the thing I defined below called `extend` could be called `is_extension_along_yoneda`? I'm terrible at naming though.

Reply:
yup, sound like good suggestions to me